File: apps/backend/src/main/kotlin/com/lyra/app/forms/domain/FormId.kt
package com.lyra.app.forms.domain

import com.lyra.common.domain.BaseId
import java.util.*

data class FormId(private val id: UUID) : BaseId<UUID>(id) {
    constructor(id: String) : this(UUID.fromString(id))
    companion object {
        fun create() = FormId(UUID.randomUUID())
    }
}

File: apps/frontend/apps/lyra-app/domain/forms/FormId.ts
import { UUIDValueObject } from '@lyra/shared';

export default class FormId extends UUIDValueObject {
  private constructor(value: string) {
    super(value);
    this.ensureIsValidUuid(value);
  }

  static random(): FormId {
    return new FormId(crypto.randomUUID());
  }

  public static create(value: string): FormId {
    return new FormId(value);
  }
}
